syntax-markdown-flavor/syntax-markdown-flavor-github/src/main/java/org/xwiki/contrib/rendering/markdown/flavor/github/internal/renderer/MarkdownGithubRenderer.java

```diff
@@ -44,26 +45,29 @@
 public class MarkdownGithubRenderer extends Markdown12Renderer
 {
     @Inject
-    private MarkdownConfiguration configuration;
+    private MarkdownGithubConfiguration configuration;
 
     /**
      * Needed by MarkdownChainingRenderer to serialize wiki link references.
      */
     @Inject
-    @Named("markdown+github/1.0/link")
+    @Named("markdown/1.2/link")
     private ResourceReferenceSerializer linkReferenceSerializer;
 
     /**
      * Needed by MarkdownChainingRenderer to serialize wiki image references.
      */
     @Inject
-    @Named("markdown+github/1.0/image")
+    @Named("markdown/1.2/image")
     private ResourceReferenceSerializer imageReferenceSerializer;
 
+    /**
+     * Needed by MarkdownChainingRenderer to serialize wiki link references.
+     */
     @Override
     protected ChainingListener createXWikiSyntaxChainingRenderer(ListenerChain chain)
     {
         return new Markdown12ChainingRenderer(chain, this.linkReferenceSerializer, this.imageReferenceSerializer,
-            this.configuration);
+                (MarkdownConfiguration) this.configuration);
```

Comment:
Here we can do `extend MarkdownConfiguration` and use the configuration as is, instead of cast. Let me know if I should.

Reply:
As said above, you shouldn't have to cast since you shouldn't need your own type of configuration. You should just reuse the MarkdownConfiguration interface and improve it.